index.bs

```diff
@@ -155,28 +156,38 @@ Parsing {#parsing}
 To <dfn abstract-op local-lt="parse header">obtain a response's embedder policy</dfn> given a
 [=response=] (|response|):
 
-1.  Let |policy| be "`unsafe-none`".
-
-2.  Let |header| be the result of [=header list/getting=] `Cross-Origin-Embedder-Policy` from
+1.  Let |header| be the result of [=header list/getting=] `Cross-Origin-Embedder-Policy` from
     |response|'s [=response/header list=].
 
-3.  If |header| is not `null`:
+2.  If |header| is `null`, then return a new [=/embedder policy=].
+
+3.  Set |header| to the result of [=isomorphic decoding=] |header|.
+
+4.  Let |parsed item| be the result of executing the [$Structured Header parsing algorithm$]
+    with <var ignore>input_string</var> set to |header|, and <var ignore>header_type</var> set
+    to "`item`".
+
+    If parsing fails, then return a new [=/embedder policy=].
+
+    ISSUE(httpwg/http-extensions#662): The ASCII requirements of Structured Headers are
+    somewhat unclear to me. Do we need to check whether |header| is an [=ASCII string=] before
+    passing it into the parsing algorithm?
+
+5.  Let |value| be the first component of |parsed item|.
 
-    1.  Set |header| to the result of [=isomorphic decoding=] |header|.
+6.  Let |parameters| be the second component of |parsed item|.
 
-    2.  Let |parsed policy| be the result of executing the [$Structured Header parsing algorithm$]
-        with <var ignore>input_string</var> set to |header|, and <var ignore>header_type</var> set
-        to "`item`".
+7.  If |value| is neither "`require-corp`" nor "`unsafe-none`", then return a new
+    [=/embedder policy=].
 
-        If parsing fails, set |parsed policy| to "`unsafe-none`".
+8.  Let |endpoint| be `null`.
 
-        ISSUE(httpwg/http-extensions#662): The ASCII requirements of Structured Headers are
-        somewhat unclear to me. Do we need to check whether |header| is an [=ASCII string=] before
-        passing it into the parsing algorithm?
+9.  For each |name| with a value of |value| in |parameters:
```

Comment:
SH defines parameters as a map, so you can just grab the specific item you want. That is, something like the following should work instead of this loop.

>  If |parsed item|'s parameters["report-to"] [=map/exists=], set |endpoint| to |parsed item|'s parameters["report-to"].

Reply:
Done.